Classes/common/CDTIncrementalStore.m

```diff
@@ -0,0 +1,2409 @@
+                       [self cleanURL:self.remoteURL], replicator, state);
+            break;
+    }
+
+    CDTLogInfo(CDTREPLICATION_LOG_CONTEXT, @"%@: %@: %@: state: %@", CDTISType,
+               [self cleanURL:self.remoteURL], replicator, state);
+}
+
+/**
+ * Called whenever the replicator changes progress
+ */
+- (void)replicatorDidChangeProgress:(CDTReplicator *)replicator
+{
+    CDTLogInfo(CDTREPLICATION_LOG_CONTEXT, @"%@: %@: %@: progressed: [%@/%@]", CDTISType,
+               [self cleanURL:self.remoteURL], replicator, @(replicator.changesProcessed),
+               @(replicator.changesTotal));
+    self.progressBlock(NO, replicator.changesProcessed, replicator.changesTotal, nil);
+}
+
+/**
+ * Called when a state transition to COMPLETE or STOPPED is
+ * completed.
+ */
+- (void)replicatorDidComplete:(CDTReplicator *)replicator
+{
+    CDTLogInfo(CDTREPLICATION_LOG_CONTEXT, @"%@: %@: %@: completed", CDTISType,
+               [self cleanURL:self.remoteURL], replicator);
+    self.progressBlock(YES, 0, 0, nil);
+    self.progressBlock = nil;
+}
+
+/**
+ * Called when a state transition to ERROR is completed.
+ */
+- (void)replicatorDidError:(CDTReplicator *)replicator info:(NSError *)info
+{
+    CDTLogError(CDTREPLICATION_LOG_CONTEXT, @"%@: %@: %@: suffered error: %@", CDTISType,
+                [self cleanURL:self.remoteURL], replicator, info);
+    self.progressBlock(YES, 0, 0, info);
+    self.progressBlock = nil;
+}
+
+#pragma mark - required methods
+- (BOOL)loadMetadata:(NSError **)error
+{
+    if (self.databaseName) {
+        return NO;
+    }
+    if (![self initializeDatabase:error]) {
+        return NO;
+    }
+    if (![self setupIndexes:error]) {
+        return NO;
+    }
+    NSDictionary *metaData = [self getMetaDataFromDocID:CDTISMetaDataDocID error:error];
+    if (!metaData) {
+        return NO;
+    }
+    if (![self checkMetaData:metaData error:error]) {
+        [NSException raise:CDTISException format:@"failed metaData check"];
+    }
+    // go directly to super
+    [super setMetadata:metaData];
+
+    // this class only exists in iOS
+    Class frc = NSClassFromString(@"NSFetchedResultsController");
+    if (frc) {
+// If there is a cache for this, it is likely stale.
+// Sadly, we do not know the name of it, so we blow them all away
+#pragma clang diagnostic ignored "-Wundeclared-selector"
+        [frc performSelector:@selector(deleteCacheWithName:) withObject:nil];
+#pragma clang diagnostic pop
+    }
+
+    return YES;
+}
+
+/**
+ *  Create a dictionary of query options for the fetch query
+ *
+ *  @param fetchRequest fetchRequest
+ *
+ *  @return options dic
+ */
+- (NSDictionary *)processOptions:(NSFetchRequest *)fetchRequest error:(NSError **)error
+{
+    NSError *err = nil;
+    NSMutableDictionary *sdOpts = [NSMutableDictionary dictionary];
+
+    NSArray *sds = [fetchRequest sortDescriptors];
+    if (sds.count) {
+        if (sds.count > 1) {
+            oops(@"not sure what to do here");
```

Comment:
@mkistler asks, does coundant sync limit requests to 100 results like the cloudantDB does.
There is indeed core data support for a fetch/limit/window. should tie these together.